### ISSUE TITLE
 RFF: Add an extend-method to have external modules hook into annotations.

### ISSUE THREAD

**Issue by Firesphere** [silverleague/silverstripe-ideannotator#69]
Modules, like the fluent module, hook into the build process as well. Therefore, causing the annotator to create unwanted items, like Content_FR, because Fluent "magically" creates those fields.
A hook, to prevent these magical or virtual fields from bleeding into the annotator would be useful.

As discussed in the previous topic. We prefer to not make exceptions for modules, but have the module carry the exception.

Said hook, would preferably be something like a `$this->extend('method', $this->owner);` kind of implementation.

ping: @lekoala 

p.s.:
RFF === Request For Feedback.

**Comment by lekoala** [comment]
I'll send you an updated PR when I find some time to do it :-)

**Comment by axyr** [comment]
Besides the fact that we might need some extension hooks, this is a bug.

The fields of DataExtensions are added to the extended DataObject.

This should not happen, because these fields are discovered by the @mixin tag ....

**Comment by lekoala** [comment]
It's probably due to the way Fluent augment the fields of the dataobject magically. These fields are not in the DataExtension itself, they are added on the fly to the $db var.

I also noticed a problem very similar to #68 where fields with locale name in it (for instance Content_fr_FR and Content_en_US) are not refreshed properly unless I remove them manually.

**Comment by wernerkrauss** [comment]
Same with translatable dataobject module 
https://github.com/bummzack/translatable-dataobject

Would be fine, if those autogenerated fields are NOT in the annotations list, as normally we don't need them and have a getter for the current locale.

IF we need it.. would a switch in the annotated class (like $include_magic_fields) that fluent and translatableDO can check be an option?

**Comment by axyr** [comment]
They should not appear in the generated annotations. its a bug in ideannotator.
Currently i'm a bit busy, but its on my radar

**Comment by Firesphere** [comment]
I'll see if I can have a go this week. But also kinda busy. I think just getting the actual fields instead of the full set might work?
But that's just guessing. Need to check before I can say anything useful about this.

**Comment by Firesphere** [comment]
ping @robbieaverill, you know Fluent a lot better than me, so you could possibly shed some light on this issue

**Comment by Firesphere** [comment]
Resolved in #107, if the provided solution doesn't give a working endpoint, please open a new issue once that one is merged in.
Closing this for now.
